File: backend/platform-core/src/main/java/com/commonsengine/platform/ai/AiFallbacks.java
package com.commonsengine.platform.ai;

import com.commonsengine.platform.ai.AiServiceDtos.ChatReply;
import com.commonsengine.platform.ai.AiServiceDtos.DispatchResult;
import com.commonsengine.platform.ai.AiServiceDtos.DispatchSuggestion;
import com.commonsengine.platform.ai.AiServiceDtos.ModerationCategory;
import com.commonsengine.platform.ai.AiServiceDtos.ModerationDecision;
import com.commonsengine.platform.ai.AiServiceDtos.ModerationResult;

import java.util.List;

/**
 * Static fallback values used when the AI service is unavailable or fails.
 */
public final class AiFallbacks {

    private AiFallbacks() {
        throw new AssertionError("Utility class — do not instantiate");
    }

    /**
     * Fallback chat reply used when the chat service is unavailable.
     */
    public static ChatReply chatFallback(String sessionId) {
        return new ChatReply(
                sessionId,
                "I'm sorry, I'm having trouble connecting to the service right now. Please try again shortly.",
                "fallback"
        );
    }

    /**
     * Fallback moderation result — conservative {@code FLAG} decision.
     */
    public static ModerationResult moderationFallback() {
        return new ModerationResult(
                ModerationDecision.FLAG,
                List.of(ModerationCategory.NONE),
                0.0
        );
    }

    /**
     * Fallback dispatch result — empty suggestions; callers should fall back
     * to the rule-based matching engine.
     */
    public static DispatchResult dispatchFallback() {
        return new DispatchResult(List.of());
    }

    /**
     * Build a simple fallback dispatch suggestion for a single worker.
     */
    public static DispatchSuggestion singleFallbackSuggestion(String workerId) {
        return new DispatchSuggestion(workerId, 0.5, "fallback — rule-based selection unavailable");
    }
}

File: backend/platform-core/src/main/kotlin/com/commonsengine/platform/ai/AiFallbacks.kt
package com.commonsengine.platform.ai

/**
 * AI 服务层降级策略 — 当 Python 服务不可用（熔断/超时/错误）时，
 * 核心业务层不应硬故障，而是退回到安全的本地兜底结果。
 *
 * 这体现架构文档「数据主权」与「反榨取」原则：
 * - 客服不可用 → 转人工（不阻塞用户）
 * - 审核不可用 → 标记待人工复审（宁可降效，不可放行违规）
 * - 调度不可用 → 返回空建议（由 Kotlin 侧 MatchingEngine 本地兜底，绝不饥饿派单）
 *
 * — Commons Engine Chief Engineer Bot（AI），#74
 */
object AiFallbacks {

    /** 客服降级：转人工。 */
    fun customerServiceFallback(userId: String?): ChatReply = ChatReply(
        reply = "智能客服暂时不可用，已为您转接人工客服，请稍候。",
        needsHuman = true,
        category = "fallback_human",
    )

    /** 内容审核降级：标记待人工复审（保守策略，不放行）。 */
    fun moderationFallback(): ModerationResult = ModerationResult(
        decision = ModerationDecision.FLAGGED,
        category = ModerationCategory.CLEAN,
        confidence = 0.0,
        reason = "AI 审核服务暂时不可用——保守标记待人工复审",
    )

    /** 调度降级：返回空建议（交由 Kotlin 侧本地匹配引擎兜底，不饥饿派单）。 */
    fun dispatchFallback(): DispatchResult = DispatchResult(
        suggestions = emptyList(),
        strategy = "fallback_empty",
    )
}
